app/lib/tenejo/preflight.rb

```diff
@@ -136,9 +136,18 @@ def self.map_header(m)
       KNOWN_HEADERS.find { |x| x.to_s.gsub(/[^0-9A-Za-z]/, '') == squashed } || m
     end
 
-    def self.read_csv(input, import_path)
+    def self.read_csv(filename, import_path = DEFAULT_UPLOAD_PATH)
+      stream = File.open(filename)
       begin
-        csv = CSV.open(input, headers: true, return_headers: true, skip_blanks: true,
+        process_csv(stream, import_path)
+      ensure
+        stream.close
+      end
+    end
+
+    def self.process_csv(input, import_path = DEFAULT_UPLOAD_PATH)
```

Comment:
Active Storage makes it easier to access the contents of the uploaded CSV as a string than as a file.  So I created an entry point into the preflight processing that allowed me to supply the string with the full CSV instead of a filename.  This did make rubocop yell at me for making the Tenejo::Preflight class too long.